src/tools/spin/spinquic.cpp

```diff
@@ -39,14 +24,21 @@ const QUIC_BUFFER Buffers[2] = {
     } \
 } while (0);
 
+#define EXIT_ON_NOT(x) do { \
+    if (!(x)) { \
+       printf("%s:%d !'%s' !\n", __FILE__, __LINE__, #x); \
+       exit(10); \
+    } \
+} while (0);
+
 template<typename T>
-T& GetRandomFromVector(std::vector<T> &vec) {
-    return vec.at(rand() % vec.size());
+T GetRandom(T UpperBound) {
+    return (T)(rand() % (int)UpperBound);
```

Comment:
Shouldn't this use QuicGetRandom()?

Reply:
I had tried and I was seeing weird results. I'm going to leave it for now.

Reply:
I see that reproducibility is needed from the RNG, so `rand()` makes sense.